### ISSUE TITLE
[BUG] `UseAzureMonitor` collects too many internal logs

### ISSUE THREAD

**Issue by TaoChenOSU** [Azure/azure-sdk-for-net#43286]
### Library name and version

Azure.Monitor.OpenTelemetry.AspNetCore 1.2.0-beta.2

### Query/Question

When I enable telemetry by doing
```
services.AddOpenTelemetry().UseAzureMonitor();
```
, my application outputs a bunch of logs from the `Azure.*` namespace (`Azure.Identity` for example) that were not previously output.

I know I can configure the log level but I'd like to know what `UseAzureMonitor` does underneath that enables extra logging, so that my app doesn't generate an excessive amount of logs going forward.

### Environment

OS: Windows
.Net: 8.0.202

**Comment by TimothyMothra** [comment]
Hi @TaoChenOSU, 
Yes, our package subscribes to all "Azure.*" traces. This should provide a complete picture of your application. 

https://github.com/Azure/azure-sdk-for-net/blob/bb90af89e27be84541e142a4ffd648825a2fdbe2/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs#L109-L110

Are you asking to disable this?

**Comment by TaoChenOSU** [comment]
Hi @TimothyMothra,

Thank you for your quick response!

Yes, I think it'd be nice to be able to disable that since these entities that are producing telemetry may not be of our concern during the prototyping stage. Plus, we can always add `Azure.*` back via `ConfigureOpenTelemetryTracerProvider`.

**Comment by TimothyMothra** [comment]
Hi @TaoChenOSU one follow up question. Sorry, the vocabulary is a little confusing here... I just wanted to quickly clarify which part you're referring to.

- "OpenTelemetry Traces" includes things like Requests and Dependencies. This is configured via the TracerProvider.
- "OpenTelemetry Logs" are the misc everything else. .NET's implementation leverages ILogger, and the configuration includes LogLevel. These go to "Application Insights Traces" table.


When you're describing `Azure.*`, you're referring to Dependencies, correct (ie: Azure Storage)?
You mentioned "log level" in your original description so I just wanted to double check that we're discussing the same thing. :)

**Comment by TaoChenOSU** [comment]
Hi @TimothyMothra,

I may be wrong, but I believe that when you add the trace from a particular entity, you also opt in to capture the logs it produces. Correct me if I am wrong.

In my case, once I do `UseAzureMonitor`, I start seeing logs from `Azure.Identity` showing up on my terminal (our project does have a dependency on it and I am testing the project locally but with telemetry enabled), as well as in ApplicationInsights' traces table.

(I am actually not sure why the logs from `Azure.Identity` were not showing up when I didn't have `UseAzureMonitor`. I didn't configure the log level for the package or anything.)

You mentioned OPTL traces, which I think that they are valuable to have in the end-to-end transaction record and in the application map in ApplicationInsights. However, I don't want unexpected logs to start showing up everywhere.

**Comment by TaoChenOSU** [comment]
FYI, this is an example of the logs I get from `Azure.Identity`. It's showing that it's trying the credential types in the order according to this doc: https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet. 
![image](https://github.com/Azure/azure-sdk-for-net/assets/12570346/16abbc2d-cdd0-4e30-addb-50335dab6201)

I am not very interested in this information, and it buries the logs that are meaningful to me. Now, I know I can set the log level to suppress the logs, but that doesn't feel like a proper solution.

**Comment by cijothomas** [comment]
@TaoChenOSU From the screenshot shared above, the logs you are seeing are not from AzureMonitor, they are not from OpenTelemetry either! 
My best guess is - they are coming from `ConsoleLoggerProvider` that is enabled by default, as part of asp.net core! You can confirm by clearing the default providers using `ClearProviders()`.

It is not clear how enabling AzureMonitor triggered these logs.. If you can share a minimal repro app, it'll be easy to narrow down what is triggering it.. @TimothyMothra's guess that AddSource("Azure.*") is the cause is a good possibility. You can confirm that by removing azuremonitor completely, and enabling OTel yourself with AddSource("Azure.*"). It is possible that Azure.identity emits ILogger logs, when tracing is enabled.

**Comment by TaoChenOSU** [comment]
Hi @cijothomas,

Thanks for your reply!

Here is a minimal repro app: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Demo

If you just run the app, you will see a bunch of logs from `Azure.Core`. But if you remove the line `builder.Services.AddOpenTelemetry().UseAzureMonitor();`, those logs will not be printed out anymore!

I tried your first suggestion, but it didn't clear out the logs unfortunately.

I also tried enabling OTel without `AzureMonitor`. This is how I did it:
```
builder.Services.AddOpenTelemetry()
    .WithTracing(tracing => tracing
        .AddSource("Azure.*")
        .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(builder.Environment.ApplicationName))
        .AddAspNetCoreInstrumentation()
    );

builder.Logging.AddOpenTelemetry(options =>
{
    options.AddConsoleExporter();
});
```
This didn't create logs from the `Azure.*` namespace.

This is really strange, because I'd also expect the logs to be printed out regardless of if I enable telemetry or not.

**Comment by cijothomas** [comment]
Thanks @TaoChenOSU ! I think I got some idea now...

Workaround for quickly resolving issue:
builder.Logging.ClearProviders();  // add this line just **before** UseAzureMonitor() to clear off built-in providers in your dev environment. (one of the built-in provider is what is printing these logs to console)
builder.Services.AddOpenTelemetry().UseAzureMonitor();

From the latest screenshot shared, it looks like LiveMetrics feature (added in recent beta releases, but not in 1.1.0 stable) is calling LiveMetricsService (in the background), and this is executing code path which has `ILogger` logs. @TimothyMothra can help investigating the root cause here and see what would be a fix to avoid this noise.

I'd assume that you should not face this issue in 1.1.0 release (as live metrics did not exist in that version).

>  as well as in ApplicationInsights' traces table.

Could you confirm if these extra logs from Azure.Core is ending up in ApplicationInsights TraceTelemetry table? (That might shoot up billing, and you'd want to open an Azure Support Ticket, and refer this issue in that.)

**Comment by TaoChenOSU** [comment]
Thanks @cijothomas!

I downgraded to Azure.Monitor.OpenTelemetry.AspNetCore 1.1.0 and now the `Azure.*` logs are gone!

And I can confirm that the extra logs do show up in AppInsights' trace table.
![image](https://github.com/Azure/azure-sdk-for-net/assets/12570346/68125d35-ee36-45ab-bb78-05d2728477f0)

**Comment by TimothyMothra** [comment]
I got more information. This is a side effect of the `AzureSdkLogForwarder` introduced recently https://github.com/Azure/azure-sdk-for-net/issues/41373.
This was added to collect extra diagnostic information from other Azure client SDKs.
This feature is documented here: https://learn.microsoft.com/en-us/dotnet/azure/sdk/logging#logging-using-azuremonitoropentelemetryaspnetcore

cc @lmolkova 

## Problem
This has the side effect of collecting ALL the http events (Posts and Responses) our AzureMonitor SDK generates for sending telemetry (both to AzureMonitor and LiveMetrics). 

![image](https://github.com/Azure/azure-sdk-for-net/assets/28785781/7192806b-f246-4b75-8010-5927e8d091e9)


## Workaround

- The easiest workaround is to downgrade to our latest stable Azure.Monitor.OpenTelemetry.AspNetCore 1.1.0.
  This feature is only exposed in our 1.2.0-beta versions.

- Alternatively you can configure logging levels to exclude these events. 
   In the screenshot above, the extra logs start with "Azure.Core". This same name can be added to the appsettings.json: 
    ![image](https://github.com/Azure/azure-sdk-for-net/assets/28785781/2cabc388-58e8-475f-b14e-c0db0a7c21d7)

**Comment by jakenuts** [comment]
It seems like disabling logging in the dotnet OTEL SDK is a challenge, especially when exporters add logging by default like UseAzureMonitor(). I've tried a hundred different methods to limit the logs produced by OTEL using Azure Monitor and none have worked so far. What's worse is that ActivityEvents added to Activities are also treated like log entries so even when logging is turned off with hacks, my ingestion #'s are untenable.  Does anyone have any tips or clear guidance on how to configure logging in appsettings to limit the output?

Tried:

1.) Setting Logging:OpenTelemetry:LogLevel:Default to "None". ➡️Log entries still exported
2.) Setting Logging:OpenTelemetry:LogLevel:Default to "Error".  ➡️ Lower level log entries still exported
3.) Adding a LogFilteringProcessor of type BaseProcessor<LogRecord> to options. ➡️ No effect on Azure Monitor (batch?).

Only this one seems to work and feels criminally bad:
4.) Explicitly removing the last ILoggerProvider in services immediately after UseAzureMonitor.
